flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/catalog/hive/HiveCatalogMetadataTestBase.java

```diff
@@ -60,4 +65,50 @@ public void testCreateFunctionCaseInsensitive() throws Exception {
     protected CatalogFunction createPythonFunction() {
         return new CatalogFunctionImpl("test.func1", FunctionLanguage.PYTHON);
     }
+
+    /**
+     * Convert the {@link ResolvedSchema} to a {@link Schema}, and also convert the {@link
+     * ResolvedExpression} back to its Unresolved state. This will enable direct comparison of the
+     * schema.
+     */
+    public static Schema fromResolvedSchema(ResolvedSchema resolvedSchema) {
```

Comment:
Refactored by introducing a `TestSchemaResolver` to ease this compare.

Reply:
Do you mean for method `CatalogTestUtil#checkEquals(CatalogTable t1, CatalogTable t2)`, t1/t2 may be `ResolvedCatalogTable` or `DefaultCatalogTable`.

Reply:
Please add some comments for it.

Reply:
If it only happans in Hive, can we use method `CatalogManagerMocks.createEmptyCatalogManager()
                .resolveCatalogTable()`? So that, we won't need to add a new class `TestSchemaResolver`.